File: 350-intersection-of-two-arrays-ii/intersection-of-two-arrays-ii.cpp
class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        vector<int> result;
        unordered_multiset<int> mp2 (nums2.begin(), nums2.end());

        for (auto n : nums1) {
            if (mp2.find(n) != mp2.end()) {
                result.push_back(n);
                mp2.erase(mp2.find(n));
            }
        }
        return result;
    }
};

File: 350-intersection-of-two-arrays-ii/intersection-of-two-arrays-ii.go
func intersect(nums1 []int, nums2 []int) []int {
    interSection := []int{}
    freqNums1 := map[int]int{}

    for _, n := range nums1 {
        freqNums1[n]++
    }

    for _, n := range nums2 {
        if v, ok := freqNums1[n]; ok && v > 0 {
            interSection = append(interSection, n)
            freqNums1[n]--
        }
    }

    return interSection
}